### ISSUE TITLE
aws-provider no support for ap-northeast-3 region that became generally available on 1/03/21

### ISSUE THREAD

**Issue by bchaja** [hashicorp/terraform-provider-aws#17882]
aws-provider ver 3.30 doesnt support ap-northeast-3 and error out with 'Invalid AWS Region ap-northeast-3'. The ap-north-east become open to all as of 1st/03 https://aws.amazon.com/blogs/aws/aws-asia-pacific-osaka-region-now-open-to-all-with-three-azs-more-services/

**Comment by bflad** [comment]
Hi @bchaja 👋  Thank you for submitting this.

The automatic region name validation that occurs during the Terraform AWS Provider configuration is based off the upstream [AWS Go SDK endpoints model](https://github.com/aws/aws-sdk-go/blob/master/models/endpoints/endpoints.json). When those are updated, which is most likely later today given previous AWS Go SDK timing with these types of announcements, the dependency update is made in this codebase, and released then the error will no longer occur.

You can workaround this error in the meantime with a [provider configuration using the `skip_region_validation` argument](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#skip_region_validation), e.g.

```terraform
provider "aws" {
  # ... potentially other configuration ...
  region                 = "ap-northeast-3"
  skip_region_validation = true
}
```

**Comment by snayler0** [comment]
FYI I have opened an issue with the aws-sdk-go team for this - https://github.com/aws/aws-sdk-go/issues/3805

Unfortunately for me - skip_region_validation is not an appropriate workaround due to some questionable... but necessary processes to update our code :)

Hopefully the change comes through soon though! Is it correct to assume that when the change is in the aws-sdk-go repo, it'll automatically mean a new provider version is built and made available here? (eg: ~v3.30.1?)

**Comment by bflad** [comment]
Automatic validation of the `ap-northeast-3` region has been merged and will release with version 3.31.0 of the Terraform AWS Provider, likely later today. As mentioned above, the `skip_region_validation` flag can be used to prevent the validation error on earlier versions:

```terraform
provider "aws" {
  # ... potentially other configuration ...
  region                 = "ap-northeast-3"
  skip_region_validation = true
}
```
